go/tasks/plugins/presto/execution_state.go

```diff
@@ -0,0 +1,569 @@
+	}
+
+	resourceConstraintsSpec := createResourceConstraintsSpec(ctx, tCtx, routingGroup)
+
+	allocationStatus, err := tCtx.ResourceManager().AllocateResource(ctx, routingGroup, uniqueID, resourceConstraintsSpec)
+	if err != nil {
+		logger.Errorf(ctx, "Resource manager failed for TaskExecId [%s] token [%s]. error %s",
+			tCtx.TaskExecutionMetadata().GetTaskExecutionID().GetID(), uniqueID, err)
+		return newState, errors.Wrapf(errors.ResourceManagerFailure, err, "Error requesting allocation token %s", uniqueID)
+	}
+	logger.Infof(ctx, "Allocation result for [%s] is [%s]", uniqueID, allocationStatus)
+
+	// Emitting the duration this execution has been waiting for a token allocation
+	if currentState.AllocationTokenRequestStartTime.IsZero() {
+		newState.AllocationTokenRequestStartTime = time.Now()
+	} else {
+		newState.AllocationTokenRequestStartTime = currentState.AllocationTokenRequestStartTime
+	}
+	waitTime := time.Since(newState.AllocationTokenRequestStartTime)
+	metric.ResourceWaitTime.Observe(waitTime.Seconds())
+
+	if allocationStatus == core.AllocationStatusGranted {
+		newState.Phase = PhaseQueued
+	} else if allocationStatus == core.AllocationStatusExhausted {
+		newState.Phase = PhaseNotStarted
+	} else if allocationStatus == core.AllocationStatusNamespaceQuotaExceeded {
+		newState.Phase = PhaseNotStarted
+	} else {
+		return newState, errors.Errorf(errors.ResourceManagerFailure, "Got bad allocation result [%s] for token [%s]",
+			allocationStatus, uniqueID)
+	}
+
+	return newState, nil
+}
+
+func composeResourceNamespaceWithRoutingGroup(ctx context.Context, tCtx core.TaskExecutionContext) (core.ResourceNamespace, error) {
+	routingGroup, _, _, _, err := GetQueryInfo(ctx, tCtx)
+	if err != nil {
+		return "", err
+	}
+	clusterPrimaryLabel := resolveRoutingGroup(ctx, routingGroup, config.GetPrestoConfig())
+	return core.ResourceNamespace(clusterPrimaryLabel), nil
+}
+
+// This function is the link between the output written by the SDK, and the execution side. It extracts the query
+// out of the task template.
+func GetQueryInfo(ctx context.Context, tCtx core.TaskExecutionContext) (
+	routingGroup string,
+	catalog string,
+	schema string,
+	statement string,
+	err error) {
+
+	taskTemplate, err := tCtx.TaskReader().Read(ctx)
+	if err != nil {
+		return "", "", "", "", err
+	}
+
+	prestoQuery := plugins.PrestoQuery{}
+	if err := utils.UnmarshalStruct(taskTemplate.GetCustom(), &prestoQuery); err != nil {
+		return "", "", "", "", err
+	}
+
+	if err := validatePrestoStatement(prestoQuery); err != nil {
+		return "", "", "", "", err
+	}
+
+	routingGroup = prestoQuery.RoutingGroup
+	catalog = prestoQuery.Catalog
+	schema = prestoQuery.Schema
+	statement = prestoQuery.Statement
+
+	inputs, err := tCtx.InputReader().Get(ctx)
+	if err != nil {
+		return "", "", "", "", err
+	}
+
+	statement, routingGroup, catalog, schema, err = presto.InterpolateInputs(ctx, *inputs, statement, routingGroup, catalog, schema)
```

Comment:
We interpolate the `statement`, `routing_config`, `catalog`, `schema` with inputs from users. See the comments I wrote in the utils file. I tested this locally and it works